elastic/kibana
# [Dashboard First] Unlink from Library Action With ReferenceOrValueEmbeddable

## Summary

Counterpart / Opposite of https://github.com/elastic/kibana/pull/75098

Adds an embeddable panel action to unlink by reference embeddables from their library item, turning them into by value embeddables. This action should work with any embeddable that implements the `ReferenceOrValueEmbeddable` interface from #74302.
![Aug-12-2020 17-09-13](https://user-images.githubusercontent.com/14276393/90068419-9b8fa200-dcbe-11ea-9afd-fd76412009fb.gif)


## How to test this?

This is just one part of a much larger change, so all changes in this PR are hidden behind a configuration value. In order for the 'by value' workflow to be made default, some architectural changes need to be completed including https://github.com/elastic/kibana/pull/67931, https://github.com/elastic/kibana/issues/71499, and https://github.com/elastic/kibana/issues/61663

1. you'll need to set the `allowByValueEmbeddables` config value to `true`:
https://github.com/elastic/kibana/blob/5b64a4cdfea92639ddf9e23c86c43dda5ea3ac68/src/plugins/dashboard/config.ts#L23

2. There are no production ready embeddables that implement the `ReferenceOrValueEmbeddable` yet so you will have to start the developer examples. `yarn start --run-examples`

3. Open a dashboard, and use the add panels functionality to add a Book embeddable. You should be able to then unlink it from its library item.

### Checklist


### For maintainers
